api/client/registry/login.go

```diff
@@ -52,7 +53,16 @@ func runLogin(dockerCli *client.DockerCli, opts loginOptions) error {
 
 	var serverAddress string
 	var isDefaultRegistry bool
-	if opts.serverAddress != "" {
+
+	// when a user supplied registry url is the default docker.io registry just ignore it
+	// without this check the login will succed, but the credentials will not be saved properly and can't be used
+	// handle urls without schema for backward compatibility and pass the integration tests
+	u, _ := url.Parse(opts.serverAddress)
+	if u.Host == "" {
+		u, _ = url.Parse("http://" + opts.serverAddress)
```

Comment:
Shouldn't be an option to specify where server address working over **http** or **https**?